src/span.js

```diff
@@ -6,6 +6,88 @@ let defaultTracer = require('./default_tracer');
 const kKeyRegExp = new RegExp(/^[a-z0-9][-a-z0-9]*/);
 
 /**
+ * SpanContext represents Span state that must propagate to descendant Spans
+ * and across process boundaries.
+ *
+ * SpanContext is logically divided into two pieces: the user-level "Baggage"
+ * (see setBaggageItem and getBaggageItem) that propagates across Span
+ * boundaries and any Tracer-implementation-specific fields that are needed to
+ * identify or otherwise contextualize the associated Span instance (e.g., a
+ * <trace_id, span_id, sampled> tuple).
+ */
+export default class SpanContext {
+
+    /**
+     * Sets a key:value pair on this SpanContext that also propagates to future
+     * children of the associated Span.
+     *
+     * setBaggageItem() enables powerful functionality given a full-stack
+     * opentracing integration (e.g., arbitrary application data from a web
+     * client can make it, transparently, all the way into the depths of a
+     * storage system), and with it some powerful costs: use this feature with
+     * care.
+     *
+     * IMPORTANT NOTE #1: setBaggageItem() will only propagate baggage items to
+     * *future* causal descendants of the associated Span.
+     *
+     * IMPORTANT NOTE #2: Use this thoughtfully and with care. Every key and
+     * value is copied into every local *and remote* child of the associated
+     * Span, and that can add up to a lot of network and cpu overhead.
+     *
+     * @param {string} key
+     * @param {string} value
+     */
+    setBaggageItem(key, value) {
+        if (API_CONFORMANCE_CHECKS) {
+            if (arguments.length !== 2) {
+                throw new Error('Invalid number of arguments');
+            }
+        }
+        if (this._imp) {
+            this._imp.setBaggageItem(key, value);
+        }
+    }
+
+    /**
+     * Returns the value for a baggage item given its key.
+     *
+     * @param  {string} key
+     *         The key for the given trace attribute.
+     * @return {string}
+     *         String value for the given key, or undefined if the key does not
+     *         correspond to a set trace attribute.
+     */
+    getBaggageItem(key) {
+        if (API_CONFORMANCE_CHECKS) {
+            if (arguments.length !== 1) {
+                throw new Error('Invalid number of arguments');
+            }
+        }
+        if (this._imp) {
+            this._imp.getBaggageItem(key);
```

Comment:
This doesn't look correct. This method is supposed to return a string, but doesn't have a return for either the noop or implementation case.